packages/devtools_app/lib/src/debugger/hover.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import '../theme.dart';
+
+/// Returns the word in the [line] for the provided hover [dx] offset given
+/// the [line]'s [textStyle].
+String wordForHover(double dx, TextSpan line, TextStyle textStyle) {
+  final index = _hoverIndex(dx, line, textStyle);
+  var word = '';
+  if (index == -1) return word;
+  word = line.children[index].toPlainText();
+  word = _mergedLinkedWords(word, index, line);
+  return word;
+}
+
+/// Merges words linked by a `.`.
+///
+/// For example, hovering over `bar` in `foo.bar.baz` would return
+/// `foo.bar.baz`.
```

Comment:
nit: should this be foo.bar instead of foo.bar.baz?

Reply:
No thi sis correct.

Reply:
can you explain that rationale to me? If a user hovers over `bar`, wouldn't they want to look at the value of `bar`, not `baz`?

Reply:
I thought Chrome's behavior was to treat all things linked by a `.` as a single "block".

I just tested and that doesn't seem to be the case:
<img width="794" alt="Screen Shot 2021-03-02 at 3 16 52 PM" src="https://user-images.githubusercontent.com/1840773/109728092-68353780-7b6a-11eb-8efe-abc941498ff8.png">

I'll update the logic.